datashader/pipeline.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import, division, print_function
+
+"""
+Declarative interface to Datashader.  
+
+Provides a configurable pipeline that makes it simpler to specify
+individual stages independently from the others, Currently does not
+add any new underlying capabilities, with all functionality also
+available using the default imperative interface provided by other
+files.
+"""
+
+import param
+
+from . import transfer_functions as tf
+from . import reductions
+from . import core
+
+
+class Interpolate(param.Parameterized):
+    """
+    Parameterized function object to interpolate colors from a scalar input.
+    """
+
+    low = param.Parameter(default="lightpink", doc="""
+         Color string or tuple specifying the starting point for interpolation.""")
+
+    high = param.Parameter(default="red", doc="""
+         Color string or tuple specifying the ending point for interpolation.""")
+
+    how = param.Parameter(default="log", doc="""
+         Function object or string specifying how to map from a scalar into color space.""")
+
+    def __call__(self, agg):
+            return tf.interpolate(agg, self.low, self.high, self.how)
```

Comment:
Perhaps a more polite way to phrase your question is, "Oh, why do you do that -- I can't see the reason?".  In this case, the reason is that I find it far more readable to scan through a list of declared parameters that is nicely indented like this:

```python
low = param.Parameter(default="lightpink", doc="""
    Color string or tuple specifying the starting point for interpolation.""")

high = param.Parameter(default="red", doc="""
    Color string or tuple specifying the ending point for interpolation.""")
```

Than one where the doc strings and the code all get run together visually:

```python
low = param.Parameter(default="lightpink", doc="""
Color string or tuple specifying the starting point for interpolation.""")

high = param.Parameter(default="red", doc="""
Color string or tuple specifying the ending point for interpolation.""")
```

This convention is something that we developed gradually from working with declarative interfaces, where code is generally dominated by user-configurable parameters like these, making it important to have good formatting for such docstrings.  In this project, it's fine to remove those extra four spaces if people prefer, but there *is* a reason they are there.